// mat-input/fesm5/mat-input.js
import { ɵɵdefineInjectable, Injectable, Component, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var MatInputService = /** @class */ (function () {
    function MatInputService() {
    }
    MatInputService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    MatInputService.ctorParameters = function () { return []; };
    /** @nocollapse */ MatInputService.ngInjectableDef = ɵɵdefineInjectable({ factory: function MatInputService_Factory() { return new MatInputService(); }, token: MatInputService, providedIn: "root" });
    return MatInputService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var MatInputComponent = /** @class */ (function () {
    function MatInputComponent() {
    }
    /**
     * @return {?}
     */
    MatInputComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    MatInputComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-mat-input',
                    template: "\n    <p>\n      mat-input works!\n    </p>\n  "
                }] }
    ];
    /** @nocollapse */
    MatInputComponent.ctorParameters = function () { return []; };
    return MatInputComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var MatInputModule = /** @class */ (function () {
    function MatInputModule() {
    }
    MatInputModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [MatInputComponent],
                    imports: [],
                    exports: [MatInputComponent]
                },] }
    ];
    return MatInputModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { MatInputComponent, MatInputModule, MatInputService };
