// mat-input/bundles/mat-input.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('mat-input', ['exports', '@angular/core'], factory) :
    (global = global || self, factory(global['mat-input'] = {}, global.ng.core));
}(this, function (exports, core) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var MatInputService = /** @class */ (function () {
        function MatInputService() {
        }
        MatInputService.decorators = [
            { type: core.Injectable, args: [{
                        providedIn: 'root'
                    },] }
        ];
        /** @nocollapse */
        MatInputService.ctorParameters = function () { return []; };
        /** @nocollapse */ MatInputService.ngInjectableDef = core.ɵɵdefineInjectable({ factory: function MatInputService_Factory() { return new MatInputService(); }, token: MatInputService, providedIn: "root" });
        return MatInputService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var MatInputComponent = /** @class */ (function () {
        function MatInputComponent() {
        }
        /**
         * @return {?}
         */
        MatInputComponent.prototype.ngOnInit = /**
         * @return {?}
         */
        function () {
        };
        MatInputComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'lib-mat-input',
                        template: "\n    <p>\n      mat-input works!\n    </p>\n  "
                    }] }
        ];
        /** @nocollapse */
        MatInputComponent.ctorParameters = function () { return []; };
        return MatInputComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var MatInputModule = /** @class */ (function () {
        function MatInputModule() {
        }
        MatInputModule.decorators = [
            { type: core.NgModule, args: [{
                        declarations: [MatInputComponent],
                        imports: [],
                        exports: [MatInputComponent]
                    },] }
        ];
        return MatInputModule;
    }());

    exports.MatInputComponent = MatInputComponent;
    exports.MatInputModule = MatInputModule;
    exports.MatInputService = MatInputService;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
